src/browserClip.ts

```diff
@@ -248,3 +248,62 @@ module.exports = (params: ClipParams) => {
     );
   }
 };
+
+const totalPatternDuration = (pattern: any) => {
+  return recursivelyApplyPatternToDurations(
+    expandStr(pattern),
+    Tone.Ticks('4n').toSeconds()
```

Comment:
Since you had add the improvement to infer correct note duration in an earlier commit, do you think we should set "4n" as a backup in case no subdiv or dur is provided by the user, for eg https://github.com/scribbletune/scribbletune/blob/master/src/clip.ts#L174

Reply:
Hey actually I missed that ! I should use here `subdiv` provided in params instead of hard-coding this subdivision value.

Reply:
Solved in https://github.com/scribbletune/scribbletune/pull/113/commits/e5a98813cc7d71cbe4d9d1af10cfb163452340f4